csFastFloat/FastDoubleParser.cs

```diff
@@ -19,24 +21,26 @@ namespace csFastFloat
 
   public static class FastDoubleParser
   {
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public  static double exact_power_of_ten(long power)
+    {
+#if NET5_0
+      Debug.Assert(power < Constants.powers_of_ten_double.Length);
+      ref double tableRef = ref MemoryMarshal.GetArrayDataReference(Constants.powers_of_ten_double);
+      return Unsafe.Add(ref tableRef, (IntPtr)power);
+#else
+      return Constants.powers_of_ten_double[power];
+#endif
 
-    public  static double exact_power_of_ten(long power) => Constants.powers_of_ten_double[power];
-
-
+    }
 
-    public static double ToFloat(bool negative, AdjustedMantissa am)
+    public static unsafe double ToFloat(bool negative, AdjustedMantissa am)
     {
-      double d;
       ulong word = am.mantissa;
       word |= (ulong)(am.power2) << DoubleBinaryConstants.mantissa_explicit_bits;
       word = negative ? word | ((ulong)(1) << DoubleBinaryConstants.sign_index) : word;
 
-      unsafe
-      {
-        Buffer.MemoryCopy(&word, &d, sizeof(double), sizeof(double));
-      }
-
-      return d;
+      return BitConverter.Int64BitsToDouble((long)word);
```

Comment:
This is even better than the previous `Unsafe.CopyBlockUnaligned` as this boils down to `vmovq xmm0, rdx` and avoid the stack-usage.

It's only possible for the 64-bit values. On the float-counterpart I used `Unsafe.Copy` now, as codegen seems a bit better (not measurable).